sinatra-contrib/lib/sinatra/capture.rb

```diff
@@ -106,12 +105,12 @@ def capture(*args, &block)
         dummy      = DUMMIES.fetch(current_engine)
         options    = { :layout => false, :locals => {:args => args, :block => block }}
 
-        buffer.try :clear
+        buffer.public_send(:clear) if buffer.respond_to?(:clear)
```

Comment:
Yes, that would be fine. I tend to pair `public_send(:meth)` with `respond_to?(:meth)` and `send(:meth)` with `respond_to?(:meth, true)` to make it clear and consistent, but in this case it's not strictly necessary.